spec/lang/well-formed.md

```diff
@@ -503,11 +531,18 @@ impl PlaceExpr {
             }
             Field { root, field } => {
                 let root = root.check_wf::<T>(locals, prog)?;
-                let (_offset, field_ty) = match root {
-                    Type::Tuple { fields, .. } | Type::Union { fields, .. } => {
+                let field_ty = match root {
+                    Type::Tuple { sized_fields, unsized_field, .. } if field == sized_fields.len() => {
```

Comment:
This logic is duplicated with the one for actually executing a field projection, I think.

Seems better to have a helper method like `get_field` or so that implements this logic just once. You might want to add a `struct TupleType` holding the fields of `Type::Tuple` to have some natural place to put this method.